### ISSUE TITLE
Obstructions <1 unit from a face break CalcPoints

### ISSUE THREAD

**Issue by ericwa** [ericwa/tyrutils-ericw#8]
Investigate the black wall face:

![spasm0003](https://cloud.githubusercontent.com/assets/239161/12992387/69c5694c-d0d1-11e5-8ad2-3c77ffcf6eee.jpg)

```
{
"spawnflags" "0"
"classname" "worldspawn"
"wad" "telefragged.wad"
{
( -112 1472 160 ) ( -112 1472 288 ) ( 192 1472 160 ) jf2brick01 0 0 0 1 1
( -112 -48 -800 ) ( 192 -48 -800 ) ( -112 -0 -800 ) jf2brick01 32 -32 0 1 1
( 192 1536 288 ) ( -112 1536 288 ) ( 192 1536 160 ) jf2brick01 0 0 0 1 1
( -640 1536 -112 ) ( -512 1536 -112 ) ( -640 1504 -112 ) jf2brick01 32 -32 0 1 1
( -896 1472 -176 ) ( -896 1600 -176 ) ( -896 1472 -160 ) jf2brick01 32 -16 0 1 1
( -832 1472 -192 ) ( -832 1344 -192 ) ( -832 1472 -64 ) jf2rust68 0 0 90 1 1
}
{
( -896 -48 -64 ) ( -896 -0 -64 ) ( -896 -48 64 ) jf2brick01 32 16 0 1 1
( 128 -0 -368 ) ( 128 -48 -368 ) ( -176 -0 -368 ) jf2plat08 0 0 0 1 1
( -704 1280 -400 ) ( -576 1280 -400 ) ( -704 1296 -400 ) jf2rust76 0 0 0 1 1
( -768 1440 -368 ) ( -768 1440 -240 ) ( -896 1440 -368 ) jf2brick01 -32 16 0 1 1
( -736 1264 -400 ) ( -736 1280 -400 ) ( -736 1264 -528 ) jf2rust68 16 0 90 1 1
( -800 1248 -400 ) ( -928 1248 -400 ) ( -800 1248 -272 ) jf2brick01 -32 16 0 1 1
}
{
( -112 1472 160 ) ( -112 1472 288 ) ( 192 1472 160 ) jf2rust68 0 0 0 1 1
( -112 -48 -800 ) ( 192 -48 -800 ) ( -112 -0 -800 ) jf2brick01 32 -32 0 1 1
( 192 1536 288 ) ( -112 1536 288 ) ( 192 1536 160 ) jf2brick01 0 0 0 1 1
( -640 1536 -112 ) ( -512 1536 -112 ) ( -640 1504 -112 ) jf2brick01 32 -32 0 1 1
( -832 1472 -192 ) ( -832 1472 -64 ) ( -832 1344 -192 ) jf2brick01 32 -16 0 1 1
( -800 1472 -176 ) ( -800 1344 -176 ) ( -800 1472 -48 ) jf2rust68 0 0 90 1 1
}
{
( -112 1472 160 ) ( -112 1472 288 ) ( 192 1472 160 ) jf2brick01 0 0 0 1 1
( -112 -48 -800 ) ( 192 -48 -800 ) ( -112 -0 -800 ) jf2brick01 0 0 0 1 1
( 192 1536 288 ) ( -112 1536 288 ) ( 192 1536 160 ) jf2brick01 0 0 0 1 1
( -896 1472 -176 ) ( -896 1472 -160 ) ( -896 1600 -176 ) jf2brick01 0 0 0 1 1
( -960 1472 -208 ) ( -960 1472 -80 ) ( -960 1344 -208 ) jf2brick01 0 0 0 1 1
( -928 1536 -112 ) ( -944 1536 -112 ) ( -928 1664 -112 ) jf2brick01 0 0 0 1 1
}
{
( -368 1248 -64 ) ( -368 1248 64 ) ( -64 1248 -64 ) jf2brick01 0 0 0 1 1
( -368 -48 -656 ) ( -64 -48 -656 ) ( -368 -0 -656 ) jf2brick01 0 0 0 1 1
( -64 1440 64 ) ( -368 1440 64 ) ( -64 1440 -64 ) jf2brick01 0 0 0 1 1
( -992 1216 -248 ) ( -992 1216 -120 ) ( -992 1088 -248 ) jf2brick01 0 0 0 1 1
( -896 1216 -288 ) ( -896 1088 -288 ) ( -896 1216 -160 ) jf2brick01 0 0 0 1 1
( -928 1200 -368 ) ( -928 1168 -368 ) ( -944 1184 -368 ) jf2plat08 0 0 0 1 1
}
{
( -960 -48 -64 ) ( -960 -0 -64 ) ( -960 -48 64 ) jf2rust68 0 0 90 1 1
( 128 1472 64 ) ( -176 1472 64 ) ( 128 1472 -64 ) jf2rust68 0 0 90 1 1
( 128 -0 -368 ) ( 128 -48 -368 ) ( -176 -0 -368 ) jf2rust68 0 0 90 1 1
( -704 1280 -400 ) ( -576 1280 -400 ) ( -704 1296 -400 ) jf2rust68 0 0 90 1 1
( -736 1264 -400 ) ( -736 1280 -400 ) ( -736 1264 -528 ) jf2rust68 0 0 90 1 1
( -784 1440 -400 ) ( -912 1440 -400 ) ( -784 1440 -272 ) jf2rust68 0 0 90 1 1
}
{
( -112 1472 160 ) ( -112 1472 288 ) ( 192 1472 160 ) jf2brick01 0 0 0 1 1
( -112 -48 -800 ) ( 192 -48 -800 ) ( -112 -0 -800 ) jf2brick01 32 -32 0 1 1
( 192 1536 288 ) ( -112 1536 288 ) ( 192 1536 160 ) jf2brick01 0 0 0 1 1
( -640 1536 -112 ) ( -512 1536 -112 ) ( -640 1504 -112 ) jf2brick01 32 -32 0 1 1
( -672 1472 -272 ) ( -672 1344 -272 ) ( -672 1472 -144 ) jf2rust68 0 0 90 1 1
( -800 1472 -176 ) ( -800 1472 -48 ) ( -800 1344 -176 ) jf2brick01 32 -16 0 1 1
}
{
( 128 1472 64 ) ( -176 1472 64 ) ( 128 1472 -64 ) jf2brick01 -32 16 0 1 1
( -704 -0 64 ) ( -704 -0 -64 ) ( -704 -48 64 ) jf2rust68 16 0 90 1 1
( 128 -0 -368 ) ( 128 -48 -368 ) ( -176 -0 -368 ) jf2rust68 0 0 0 1 1
( -704 1280 -400 ) ( -576 1280 -400 ) ( -704 1296 -400 ) jf2rust68 0 0 0 1 1
( -784 1376 -368 ) ( -912 1376 -368 ) ( -784 1376 -240 ) jf2brick01 -32 16 0 1 1
( -736 1264 -400 ) ( -736 1264 -528 ) ( -736 1280 -400 ) jf2brick01 32 16 0 1 1
}
{
( -368 1216 -64 ) ( -368 1216 64 ) ( -64 1216 -64 ) jf2brick01 32 16 0 1 1
( -960 1216 -248 ) ( -960 1088 -248 ) ( -960 1216 -120 ) jf2brick01 32 16 0 1 1
( -992 1376 -240 ) ( -992 1392 -240 ) ( -992 1376 -112 ) jf2brick01 32 16 0 1 1
( -960 1344 -272 ) ( -960 1360 -272 ) ( -832 1344 -272 ) jf2plat08 0 0 90 1 1
( -976 1472 -272 ) ( -976 1472 -144 ) ( -1104 1472 -272 ) jf2rust72 0 0 0 1 1
( -976 1200 -372 ) ( -976 1216 -372 ) ( -992 1200 -372 ) jf2brick01 32 -32 0 1 1
}
{
( -368 1216 -64 ) ( -368 1216 64 ) ( -64 1216 -64 ) jf2rust68 16 0 90 1 1
( -64 1472 64 ) ( -368 1472 64 ) ( -64 1472 -64 ) jf2lite10 0 16 0 1 1
( -64 -0 -240 ) ( -64 -48 -240 ) ( -368 -0 -240 ) jf2rust68 0 0 0 1 1
( -960 1216 -248 ) ( -960 1088 -248 ) ( -960 1216 -120 ) jf2rust68 16 0 90 1 1
( -992 1376 -240 ) ( -992 1392 -240 ) ( -992 1376 -112 ) jf2rust68 0 0 0 1 1
( -960 1344 -272 ) ( -832 1344 -272 ) ( -960 1360 -272 ) jf2rust68 0 0 0 1 1
}
{
( -112 1472 160 ) ( -112 1472 288 ) ( 192 1472 160 ) jf2brick01 0 0 0 1 1
( 192 1536 288 ) ( -112 1536 288 ) ( 192 1536 160 ) jf2rust68 16 0 90 1 1
( -640 1536 -80 ) ( -512 1536 -80 ) ( -640 1504 -80 ) jf2brick01 0 0 0 1 1
( -896 1472 -176 ) ( -896 1472 -160 ) ( -896 1600 -176 ) jf2brick01 0 0 0 1 1
( -960 1472 -208 ) ( -960 1472 -80 ) ( -960 1344 -208 ) jf2brick01 0 0 0 1 1
( -928 1536 -112 ) ( -928 1664 -112 ) ( -944 1536 -112 ) jf2brick01 0 0 0 1 1
}
{
( -368 -48 -96 ) ( -64 -48 -96 ) ( -368 -0 -96 ) jf2rust76 0 0 0 1 1
( -832 -0 432 ) ( -832 -0 304 ) ( -832 -48 432 ) jf2brick01 32 0 0 1 1
( -928 1152 -48 ) ( -928 1024 -48 ) ( -944 1152 -48 ) jf2brick01 32 -32 0 1 1
( -912 1216 -112 ) ( -912 1216 -240 ) ( -944 1216 -112 ) jf2brick01 32 0 0 1 1
( -960 1472 -112 ) ( -960 1472 -240 ) ( -944 1472 -112 ) jf2brick01 32 0 0 1 1
( -1248 1280 -112 ) ( -1248 1280 -240 ) ( -1248 1292 -112 ) jf2brick01 32 0 0 1 1
}
{
( -112 1472 160 ) ( -112 1472 288 ) ( 192 1472 160 ) jf2brick01 0 0 0 1 1
( -640 -0 288 ) ( -640 -0 160 ) ( -640 -48 288 ) jf2rust68 32 -16 0 1 1
( 192 -0 128 ) ( 192 -48 128 ) ( -112 -0 128 ) jf2brick01 32 -32 0 1 1
( -640 1536 -112 ) ( -640 1504 -112 ) ( -512 1536 -112 ) jf2brick01 32 -32 0 1 1
( -896 1472 -176 ) ( -896 1600 -176 ) ( -896 1472 -160 ) jf2brick01 32 -16 0 1 1
( -512 1504 -16 ) ( -512 1504 112 ) ( -640 1504 -16 ) jf2brick01 32 -16 0 1 1
}
{
( -176 1216 304 ) ( -176 1216 432 ) ( 128 1216 304 ) jf2brick01 -32 0 0 1 1
( -176 -48 -96 ) ( 128 -48 -96 ) ( -176 -0 -96 ) jf2rust76 0 0 0 1 1
( 128 1472 432 ) ( -176 1472 432 ) ( 128 1472 304 ) jf2brick01 -32 0 0 1 1
( -592 1312 -16 ) ( -560 1328 -16 ) ( -576 1312 -16 ) jf2brick01 -32 -32 0 1 1
( -800 1248 -160 ) ( -800 1248 -288 ) ( -800 1264 -160 ) jf2brick01 32 0 0 1 1
( -672 1248 -160 ) ( -672 1264 -160 ) ( -672 1248 -288 ) jf2brick01 32 0 0 1 1
}
}
{
"spawnflags" "0"
"classname" "func_detail"
{
( -992 -48 352 ) ( -992 -0 352 ) ( -992 -48 480 ) jf2rust73 0 0 0 1 1
( -336 -48 -112 ) ( -32 -48 -112 ) ( -336 -0 -112 ) jf2rust68 0 0 0 1 1
( -960 1248 -112 ) ( -960 1264 -112 ) ( -960 1248 -240 ) jf2rust73 0 0 0 1 1
( -976 1216 -112 ) ( -1104 1216 -112 ) ( -976 1216 16 ) jf2brick01 0 48 0 1 1
( -976 1472 -112 ) ( -976 1472 16 ) ( -1104 1472 -112 ) jf2brick01 0 48 0 1 1
( -992 1424 -96 ) ( -992 1408 -96 ) ( -1120 1424 -96 ) jf2brick01 0 -32 0 1 1
}
}
{
"spawnflags" "0"
"classname" "func_group"
{
( -960 1744 -128 ) ( -960 1744 -112 ) ( -960 1856 -128 ) jf2rust01 0 32 0 1 1
( -1008 1472 -128 ) ( -1200 1472 -128 ) ( -1008 1472 -112 ) jf2rust68 0 0 0 1 1
( -1008 1744 -240 ) ( -1008 1856 -240 ) ( -1200 1744 -240 ) jf2rust01 0 -64 90 1 1
( -992 1856 -112 ) ( -992 1744 -112 ) ( -992 1856 -128 ) jf2rust72 0 -32 0 1 1
( -1200 1488 -112 ) ( -1200 1488 -128 ) ( -1008 1488 -112 ) jf2rust01 0 16 180 1 -1
( -992 1472 -192 ) ( -976 1472 -192 ) ( -992 1344 -192 ) jf2rust01 0 -64 90 1 1
}
{
( -960 1744 -128 ) ( -960 1744 -112 ) ( -960 1856 -128 ) jf2rust01 0 32 0 1 1
( -1008 1472 -128 ) ( -1200 1472 -128 ) ( -1008 1472 -112 ) jf2rust68 0 0 0 1 1
( -992 1856 -112 ) ( -992 1744 -112 ) ( -992 1856 -128 ) jf2lite10 0 0 0 1 1
( -1200 1488 -112 ) ( -1200 1488 -128 ) ( -1008 1488 -112 ) jf2rust01 0 16 180 1 -1
( -976 1472 -160 ) ( -976 1344 -160 ) ( -992 1472 -160 ) jf2rust01 0 -64 90 1 1
( -992 1472 -192 ) ( -992 1344 -192 ) ( -976 1472 -192 ) jf2rust01 0 -64 90 1 1
}
{
( -960 1744 -128 ) ( -960 1744 -112 ) ( -960 1856 -128 ) jf2rust01 0 32 0 1 1
( -1008 1472 -128 ) ( -1200 1472 -128 ) ( -1008 1472 -112 ) jf2rust68 0 0 0 1 1
( -992 1856 -112 ) ( -992 1744 -112 ) ( -992 1856 -128 ) jf2rust72 0 -32 0 1 1
( -1200 1488 -112 ) ( -1200 1488 -128 ) ( -1008 1488 -112 ) jf2rust01 0 16 180 1 -1
( -1200 1856 -112 ) ( -1008 1856 -112 ) ( -1200 1744 -112 ) jf2rust01 0 -64 90 1 1
( -976 1472 -160 ) ( -992 1472 -160 ) ( -976 1344 -160 ) jf2rust01 0 -64 90 1 1
}
}
{
"spawnflags" "0"
"classname" "func_detail"
{
( -656 1344 -272 ) ( -656 1344 -240 ) ( -496 1344 -272 ) jf2plat04 -16 0 0 1 1
( -496 1360 -244 ) ( -512 1376 -244 ) ( -656 1360 -244 ) {plat07 32 0 0 1 1
( -496 1360 -240 ) ( -656 1360 -240 ) ( -512 1376 -240 ) {plat07 32 32 0 1 1
( -959 1344 -244 ) ( -960 1344 -240 ) ( -959 1216 -244 ) jf2plat04 16 0 0 1 1
( -384 1471 -244 ) ( -256 1471 -244 ) ( -384 1472 -240 ) jf2plat04 -16 0 0 1 1
( -385 1216 -244 ) ( -385 1088 -244 ) ( -384 1216 -240 ) jf2plat04 16 0 0 1 1
}
{
( -528 1368 -240 ) ( -592 1368 -240 ) ( -528 1368 -256 ) jf2grey10 0 -16 0 1 1
( -944 1368 -240 ) ( -944 1360 -240 ) ( -944 1368 -256 ) jf2grey10 16 -16 0 1 1
( -528 1360 -240 ) ( -528 1360 -256 ) ( -592 1360 -240 ) jf2grey10 0 -16 0 1 1
( -528 1368 -245 ) ( -528 1360 -245 ) ( -592 1368 -245 ) jf2grey10 0 -16 0 1 1
( -528 1368 -256 ) ( -592 1368 -256 ) ( -528 1360 -256 ) jf2grey10 0 -16 0 1 1
( -352 1360 -240 ) ( -352 1368 -240 ) ( -352 1360 -256 ) jf2grey10 16 -16 0 1 1
}
}
{
"spawnflags" "0"
"classname" "func_illusionary"
{
( -832 -48 352 ) ( -832 -0 352 ) ( -832 -48 480 ) jf2rust73 0 0 0 1 1
( -176 -48 -112 ) ( 128 -48 -112 ) ( -176 -0 -112 ) jf2rust68 0 0 0 1 1
( -800 1248 -112 ) ( -800 1264 -112 ) ( -800 1248 -240 ) jf2rust73 0 0 0 1 1
( -816 1216 -112 ) ( -944 1216 -112 ) ( -816 1216 16 ) jf2brick01 -32 48 0 1 1
( -816 1472 -112 ) ( -816 1472 16 ) ( -944 1472 -112 ) jf2brick01 -32 48 0 1 1
( -832 1424 -96 ) ( -832 1408 -96 ) ( -960 1424 -96 ) jf2brick01 -32 -32 0 1 1
}
{
( -818 1472 -142 ) ( -818 1472 -136 ) ( -818 1420 -112 ) jf2rust62 16 -16 180 1 -1
( -814 1472 -142 ) ( -814 1420 -112 ) ( -814 1472 -136 ) jf2rust62 16 -16 180 1 -1
( -818 1420 -112 ) ( -814 1420 -112 ) ( -818 1472 -142 ) jf2rust62 32 -16 -180 1 1
( -814 1472 -136 ) ( -814 1431 -112 ) ( -818 1472 -136 ) jf2rust62 32 -16 -180 1 1
( -818 1431 -112 ) ( -814 1431 -112 ) ( -818 1420 -112 ) jf2rust62 32 -16 -180 1 1
( -818 1472 -142 ) ( -814 1472 -142 ) ( -818 1472 -136 ) jf2rust62 32 -16 -180 1 -1
}
}
{
"spawnflags" "0"
"classname" "func_detail"
{
( -752 1464 -80 ) ( -640 1464 -80 ) ( -752 1464 -128 ) jf2lite02 0 -32 -90 1 1
( -752 1456 -96 ) ( -752 1472 -96 ) ( -640 1456 -96 ) jf2rust69 0 -32 90 1 -1
( -784 1456 -80 ) ( -784 1456 -128 ) ( -784 1472 -80 ) jf2rust69 0 -32 0 1 1
( -640 1472 -128 ) ( -640 1472 -80 ) ( -752 1472 -128 ) jf2rust69 0 -32 -90 1 1
( -640 1472 -128 ) ( -752 1472 -128 ) ( -640 1456 -128 ) jf2rust69 0 -32 90 1 -1
( -688 1472 -128 ) ( -688 1456 -128 ) ( -688 1472 -80 ) jf2rust69 0 -32 0 1 1
}
}
{
"spawnflags" "0"
"classname" "light"
"origin" "-752 1456 -112"
"angle" "165"
}
{
"spawnflags" "0"
"classname" "light"
"origin" "-720 1456 -112"
"angle" "165"
}
{
"spawnflags" "0"
"classname" "light"
"origin" "-784 1232 -260"
"angle" "180"
}
{
"spawnflags" "0"
"classname" "light"
"origin" "-784 1232 -292"
"angle" "180"
}
{
"spawnflags" "0"
"classname" "light"
"origin" "-840 1184 -360"
}
{
"spawnflags" "0"
"classname" "info_player_start"
"origin" "-800 1296 -344"
"angle" "90"
}
```

**Comment by ericwa** [comment]
Ahhhh! This is tripping up CalcPoints. The wall face isn't split by the walkway, but the walkway blocks the traceline from the wall face's midpoint to the sample points. Should probably switch to a better CalcPoints implementation that uses Mod_PointInLeaf

**Comment by ericwa** [comment]
Fixed in a439f891a1971815ce1357658f89936aafa1b56d
